services/shhext/api_test.go

```diff
@@ -52,3 +56,52 @@ func TestMessagesRequest_setDefaults(t *testing.T) {
 		})
 	}
 }
+
+func TestTopicsToBloom(t *testing.T) {
+	t1 := stringToTopic("t1")
+	b1 := whisper.TopicToBloom(t1)
+	t2 := stringToTopic("t2")
+	b2 := whisper.TopicToBloom(t2)
+	t3 := stringToTopic("t3")
+	b3 := whisper.TopicToBloom(t3)
+
+	reqBloom := topicsToBloom(t1)
+	assert.True(t, whisper.BloomFilterMatch(reqBloom, b1))
+	assert.False(t, whisper.BloomFilterMatch(reqBloom, b2))
+	assert.False(t, whisper.BloomFilterMatch(reqBloom, b3))
+
+	reqBloom = topicsToBloom(t1, t2)
+	assert.True(t, whisper.BloomFilterMatch(reqBloom, b1))
+	assert.True(t, whisper.BloomFilterMatch(reqBloom, b2))
+	assert.False(t, whisper.BloomFilterMatch(reqBloom, b3))
+
+	reqBloom = topicsToBloom(t1, t2, t3)
+	assert.True(t, whisper.BloomFilterMatch(reqBloom, b1))
+	assert.True(t, whisper.BloomFilterMatch(reqBloom, b2))
+	assert.True(t, whisper.BloomFilterMatch(reqBloom, b3))
+}
+
+func TestCreateBloomFilter(t *testing.T) {
+	t1 := stringToTopic("t1")
+	t2 := stringToTopic("t2")
+
+	req := MessagesRequest{Topic: t1}
+	bloom := createBloomFilter(req)
+	assert.Equal(t, topicsToBloom(t1), bloom)
+
+	req = MessagesRequest{Topics: []whisper.TopicType{t1, t2}}
+	bloom = createBloomFilter(req)
+	assert.Equal(t, topicsToBloom(t1, t2), bloom)
+}
+
+func stringToTopic(s string) whisper.TopicType {
+	h := sha3.NewKeccak256()
```

Comment:
it is not really necessary to hash it for the test. can be any byte

Reply:
@dshulyak yes you are right, I had it at the beginning for debugging but now it's not really needed. I'm going to change it.